homeassistant/components/zwave/config_flow.py

```diff
@@ -0,0 +1,85 @@
+"""Config flow to configure Z-Wave."""
+from collections import OrderedDict
+import logging
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import callback
+
+from .const import (
+    CONF_AUTOHEAL, CONF_POLLING_INTERVAL,
+    CONF_USB_STICK_PATH, CONF_NETWORK_KEY,
+    DEFAULT_CONF_AUTOHEAL, DEFAULT_CONF_USB_STICK_PATH,
+    DEFAULT_POLLING_INTERVAL, DOMAIN)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@callback
+def configured_hosts(hass):
+    """Return a set of the configured hosts."""
+    return set(entry.data[CONF_USB_STICK_PATH] for entry
+               in hass.config_entries.async_entries(DOMAIN))
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class ZwaveFlowHandler(config_entries.ConfigFlow):
+    """Handle a Z-Wave config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    def __init__(self):
+        """Initialize the Z-Wave config flow."""
+        self.usb_path = CONF_USB_STICK_PATH
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        return await self.async_step_init(user_input)
+
+    async def async_step_init(self, user_input=None):
+        """Handle a flow start."""
+        if configured_hosts(self.hass):
+            return self.async_abort(reason='one_instance_only')
+
+        if user_input is not None:
```

Comment:
We should test that the USB Stick path exists and is accessible. Bonus points if we can verify if it's a valid Z-Wave stick (can we have OpenZ-Wave check this?)